cmd/gopass-jsonapi/internal/jsonapi/responses.go

```diff
@@ -197,6 +199,35 @@ func (api *API) respondCreateEntry(ctx context.Context, msgBytes []byte) error {
 	}, api.Writer)
 }
 
+func (api *API) respondEditEntry(ctx context.Context, msgBytes []byte) error {
+	var message editEntryMessage
+	if err := json.Unmarshal(msgBytes, &message); err != nil {
+		return errors.Wrapf(err, "failed to unmarshal JSON message")
+	}
+
+	if _, err := api.Store.Get(ctx, message.Name, "latest"); err != nil {
+		return fmt.Errorf("secret %s does not exist", message.Name)
```

Comment:
Thanks for the PR @rounakdatta.

Maybe it would be even simpler to just reuse the "create" command and allow it to update existing secrets?
As far as I see, the only difference here is the error handling.

Alternatively at least the function should be re-used, instead of duplicating the logic.

What's your preferred solution?

Reply:
Sorry for getting back late @Pharb , agree to your point. I've simplified it now, and got rid of the unnecessarily introduced `editEntryMessage ` struct. Thanks, please review and recommend next steps.